src/txBaseMsgProcessor.cpp

```diff
@@ -208,43 +203,135 @@ void TxBaseMsgProcessor::ProcessTxBaseMsg(const processMempoolTx& mempoolProcess
             vEraseQueue.push_back(hashToProcess);
         } else if (dataToProcess.txBaseProcessingState == MempoolReturnValue::PARTIALLY_VALIDATED)
         {
-            //TODO: once batch processing is integrated, pull all PARTIALLY_VALIDATED txes and process them altogether
             LOCK2(cs_main, mempool.cs);
             CCoinsViewMemPool backingView(pcoinsTip, mempool);
             CCoinsViewCache supportView(&backingView);
             CScProofVerifier scVerifier{CScProofVerifier::Verification::Strict};
-            if (txToProcess.IsCertificate())
+            std::map<uint256,unsigned int> hashToIdx;
+
+            // update in-place all PARTIALLY_VALIDATED txes/certs
+            txBaseMsgQueue.push_back(dataToProcess); //Reinsert dataToProcess to process all PARTIALLY_VALIDATED txes
+            for(unsigned int idx = 0; idx < txBaseMsgQueue.size(); ++idx)
+            {
+                TxBaseMsg_DataToProcess& enqueuedItem = txBaseMsgQueue.at(idx);
+                if (enqueuedItem.txBaseProcessingState != MempoolReturnValue::PARTIALLY_VALIDATED)
+                    continue;
+
+                // Mempool may have changed since partial validation has been made. Recheck for double spends
+                if (!supportView.HaveInputs(*enqueuedItem.pTxBase))
```

Comment:
We check Certs and Txs from the queue against `supportView`, but never update the `supportView`. is it fine?
Are we sure that in the queue there are no txs/certs spending the same inputs?

Maybe we need to check this after proof validation before running `StoreCertToMempool/StoreTxToMempool`.